lib/cppcheck.cpp

```diff
@@ -1420,8 +1420,11 @@ void CppCheck::executeAddons(const std::vector<std::string>& files, const std::s
     std::string fileList;
 
     if (files.size() >= 2 || endsWith(files[0], ".ctu-info")) {
+        // TODO: can this conflict when using -j?
```

Comment:
I think we hit this issue in #6696.

Reply:
I think there's a bug with -j here.

If I run `cppcheck` with an addon and use the `-j` flag and `cppcheck` analyzes the same source file twice for any reason (could be because its compiled twice with different flags) then it dies with:

>  error: Bailing out from analysis: Checking file failed: Failed to execute addon 'misra' - exitcode is 1 [internalError]

`--verbose` shows output from the plugin like this that makes it seem like one of the xml artifact files is malformed
https://gist.github.com/andymacg/69ff6f79e01bee2c75afdb7f674762cd 
However when I manually inspected the xml file, it parsed OK, so I think its a race condition where two threads overwrite the same file.

Reply:
I made a fix with #7082. But I am not sure how this can be tested effectively.

A unit test that ensures that the PID is part of the file list name? Or do you have any other suggestion?